src/lib/getKclSamplesManifest.ts

```diff
@@ -0,0 +1,31 @@
+import { KCL_SAMPLES_MANIFEST_URLS } from './constants'
+import { isDesktop } from './isDesktop'
+
+export type KclSamplesManifestItem = {
+  file: string
+  title: string
+  description: string
+}
+
+export async function getKclSamplesManifest() {
+  let response = await fetch(KCL_SAMPLES_MANIFEST_URLS.remote)
+  if (!response.ok) {
+    console.warn(
+      'Failed to fetch latest remote KCL samples manifest, falling back to local:',
+      response.statusText
+    )
+    response = await fetch(
+      (isDesktop() ? '.' : '') + KCL_SAMPLES_MANIFEST_URLS.localFallback
```

Comment:
I don't quite understand the point of the local fallbacks? the fallback is only for the manifest file right?

Trying to think through scenarios, like if the github API is down we'll still have the manifest file, but what good is that if users can't load the files?

Reply:
You're probably right! I was getting nervous we'd be hitting that raw file too much and get rate-limited, so this isn't really to cover if all of GitHub goes down, just if we have enough users running the app to start getting blocked on the manifest

Reply:
Maybe we can tear it out in a future PR if that rate limit idea isn't worth it